src/Components/Server/src/Microsoft.AspNetCore.Components.Server.csproj

```diff
@@ -50,10 +50,14 @@
     <Compile Include="$(MessagePackRoot)SequenceReaderExtensions.cs" LinkBase="BlazorPack\MessagePack" />
   </ItemGroup>
 
-  <ItemGroup>
-    <EmbeddedResource Include="..\..\Browser.JS\dist\$(Configuration)\blazor.server.js" LogicalName="_framework/%(Filename)%(Extension)" />
+  <PropertyGroup>
+    <BlazorServerJSFile>..\..\Browser.JS\dist\Release\blazor.server.js</BlazorServerJSFile>
+    <BlazorServerJSFile Condition="'$(Configuration)' == 'Debug' AND Exists('..\..\Browser.JS\dist\Debug\blazor.server.js')">..\..\Browser.JS\dist\Debug\blazor.server.js</BlazorServerJSFile>
+  </PropertyGroup>
 
-    <EmbeddedResource Include="..\..\Browser.JS\dist\$(Configuration)\blazor.server.js.map" LogicalName="_framework/%(Filename)%(Extension)" Condition="Exists('..\..\Browser.JS\dist\$(Configuration)\blazor.server.js.map')" />
+  <ItemGroup>
+    <EmbeddedResource Include="$(BlazorServerJSFile)" LogicalName="_framework/%(Filename)%(Extension)" />
+    <EmbeddedResource Include="$(BlazorServerJSFile).map" LogicalName="_framework/%(Filename)%(Extension)" Condition="Exists('$(BlazorServerJSFile).map')" />
```

Comment:
I'd be interested to know why these changes were needed. No doubt it's for a good reason, but it's not clear to me on the surface why the previous logic was inadequate.

Reply:
When I switched the package.json to use `link`, the Debug builds of blazor.*.js had absolute paths to SignalR embedded in them. The production builds did not since it does not include line mappings. I tried a couple of ways to solving it including rewriting the file after it was generated, but that didn't seem to help on the CI.

The change here is to say that production builds of the file are checked in and always available and we'd use Debug builds if they're available and we're building in Debug. What's missing is here a possible project reference to the npm proj which would help ensure building the server project in Debug would result in the debug files to be generated. But I wanted to sort out build incrementalism for the npmproj (preview7?) before we did that.